cypress/integration/seededFlows/podcastFlows/followPodcast.spec.js

```diff
@@ -10,26 +10,22 @@ describe('Follow podcast', () => {
   it('Follows and unfollows a podcast', () => {
     cy.get('[data-follow-clicks-initialized]');
 
-    cy.findByRole('heading', {
-      name: 'Developer on Fire Developer on Fire Follow',
-    });
     cy.findByRole('button', { name: 'Follow podcast: Developer on Fire' }).as(
       'followButton',
     );
 
     cy.get('@followButton').click();
     // Inner text should now be following
     cy.get('@followButton').should('have.text', 'Following');
+    cy.get('@followButton').should('have.attr', 'aria-pressed', 'true');
 
     // Check that state is persisted on refresh
     cy.visitAndWaitForUserSideEffects('/developeronfire');
-    cy.findByRole('button', { name: 'Unfollow podcast: Developer on Fire' }).as(
-      'followButton',
-    );
```

Comment:
As we refresh the page on the previous line, it's a good idea to get a fresh handle to the element here, otherwise Cypress may try to assert on the element on the previous page, which is now gone 😲  (we've seen a fair bit of test flakiness in the past because of this). This applies to a few other places below where a similar line has been deleted on page refresh.

In all such cases, I think it would be useful here to check that when the page loads again, this button still has the attribute `aria-pressed='true'`, to make sure the right following state has been initialized

Reply:
@aitchiss  `name: 'Developer on Fire Developer on Fire Follow'` keeps on toggling with `name: 'Developer on Fire Developer on Fire Following' during every `cy.visitAndWaitForUserSideEffects('/developeronfire');` which leads to tests failing

Reply:
That's so weird! I can't explain exactly why, but it seems like the issue is that we are still trying to get the follow button by the alias `@followButton` - but that particular instance of the button has disappeared (because we refreshed the page). It seems almost like Cypress is trying to replay the start of the test where we first defined the `followButton` alias 🤔  I've never seen that happen before though 🙃 

It seems to work as expected if we refresh the alias after each reload e.g.:

```js
cy.visitAndWaitForUserSideEffects('/developeronfire');
cy.findByRole('button', { name: 'Follow podcast: Developer on Fire' }).as(
  'followButton',
);
```

Reply:
@aitchiss Updated the change.